### PR TITLE
[MRG] support np.array as ica.exclude attribute

### LINKED ISSUE #5182: ica.exclude does not support numpy array
If `ica.exclude` is a numpy array it breaks here:

https://github.com/mne-tools/mne-python/blob/ca35ce9923ae9dc02fe55d7b321f71337903a81b/mne/preprocessing/ica.py#L1368

```
import mne
import numpy as np
from mne.preprocessing import ICA, create_ecg_epochs
from mne.datasets import sample

data_path = sample.data_path()
raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'

raw = mne.io.read_raw_fif(raw_fname, preload=True)
raw.pick_types(meg=True, eeg=False, exclude='bads', stim=True)
raw.filter(1, 30, fir_design='firwin')

events = mne.find_events(raw, stim_channel='STI 014')
epochs = mne.Epochs(raw, events, event_id=None, tmin=-0.2, tmax=0.5, preload=True)

ica = ICA(n_components=0.95, method='fastica').fit(epochs)

####### This is the problem
ica.exclude = [0, 1, 2]  #This Works
# ica.exclude = np.array([0, 1, 2]) # This fails

_ = ica.apply(epochs) # This Breaks
```

### PR DESCRIPTION
**Pull request by massich**
Fix #5182

### CODE REVIEW

File: mne/preprocessing/tests/test_ica.py [hunk 1]
```diff
@@ -684,6 +684,14 @@ def test_ica_reject_buffer(method):
     assert_equal(len(log), 1)
 
 
+@pytest.mark.parametrize("exclude", [np.array([0, 1]), [0, 1]])
```
[the remark this answers is not in the document]
Comment (jona-sassenhagen) [reply mne/preprocessing/tests/test_ica.py]: Just make line 445:

`for exclude in [[], [0], np.array([1, 2, 3])]:`

?

**Comment by massich** [comment]
Maybe [this comment](https://github.com/mne-tools/mne-python/pull/5183#discussion_r185712947) can be starting point for a discussion in refactoring some part of the tests. IMHO I find them really long, they test many things and makes it difficult to navigate them for a newcomer.

**Comment by larsoner** [comment]
I'm fine with the refactoring as long as it does not add much time to the tests. Even 1 second is too much for such a trivial test of functionality, this should ideally add < 100 ms to testing time.

If it adds more than that, you either have to trim it down (reduce channels, time samples, and fitting iterations) until it does. Or you could do what @jona-sassenhagen suggested, since it probably would add less time as the `fit` is probably the slow step.

**Comment by agramfort** [comment]
@massich can you rebase? thx

**Comment by agramfort** [comment]
conflict fixed

**Comment by massich** [comment]
Thx for rebasing

File: mne/preprocessing/tests/test_ica.py [hunk 2]
```diff
@@ -442,13 +442,14 @@ def test_ica_additional(method):
...
-        for exclude in [[], [0]]:
+        for exclude in [[], [0], np.array([1, 2, 3])]:
             ica.exclude = exclude
             ica.labels_ = {'foo': [0]}
             ica.save(test_ica_fname)
             ica_read = read_ica(test_ica_fname)
-            assert_true(ica.exclude == ica_read.exclude)
+            assert_true(list(ica.exclude) == ica_read.exclude)
             assert_equal(ica.labels_, ica_read.labels_)
+            ica.apply(raw)
```
Comment (jona-sassenhagen) [inline mne/preprocessing/tests/test_ica.py]: This line is necessary?

**Comment by massich** [comment]
Yes otherwise we don't fit with a np.array 'cos it gets reassign to []

File: mne/preprocessing/tests/test_ica.py (hunk 2 above)
Comment (agramfort) [reply mne/preprocessing/tests/test_ica.py]: yes it's the non-regression test needed.

**Comment by agramfort** [comment]
thx @massich

**Comment by jona-sassenhagen** [comment]
Thanks @massich !

**Pull request merged**